lib/rspec-puppet/adapters.rb

```diff
@@ -194,6 +197,21 @@ def settings_map
     end
 
     class Adapter6X < Adapter40
+      def setup_puppet(example_group)
+        super
+
+        case RSpec.configuration.facter_implementation.to_sym
+        when :rspec
+          raise "Runtime implementations are not supported in Puppet #{Puppet.version}" unless Puppet.respond_to?(:runtime)
```

Comment:
I think this may not be sufficient. Would this be needed to support Puppet < 6.25 but with `Puppet.runtime`?
```suggestion
          raise "Runtime implementations are not supported in Puppet #{Puppet.version}" unless Puppet.respond_to?(:runtime) && Puppet.runtime.key?(:facter)
```

Reply:
Looks like `Puppet.runtime` was added in https://github.com/puppetlabs/puppet/commit/a5109e4dd0fd10750a6eea8f6fe96b5c7d509011 (6.11.0), while the Facter implementation was added in 6.25.0.

I believe using this implementation would work on Puppet < 6.25.0 but it would provide no speed improvements since Puppet still used Facter in those versions.

We could make it more specific using your suggestion, this way we'd avoid possible confusions if users set this and notice no speed improvements (probably if they're using versions older than 6.25.0).

Reply:
@ekohl unfortunately `Puppet.runtime` is not a hash per se, so I don't see a way to check if `Puppet.runtime` has a specific service or not. We'll have to make do with the existing implementation, which should not be breaking.

Reply:
I do think the compatibility with Puppet >= 6.11 and < 6.25 would be good. There's no guarantee those aren't used and rspec-puppet is in general very broad in its compatibility (which I think is good). Please implement my suggestion :)

Reply:
Somehow I missed your reply in between.

Is there another way to detect this? If not, can we somehow get feedback to the Puppet dev team that feature detection is a useful thing and it should be considered? If not now, at least for future features.

Reply:
I just remembered that `#instance_variable_get` exists in Ruby so I can implement this 🤦🏼